plugins/elantp/fu-elantp-hid-device.c

```diff
@@ -271,6 +298,10 @@ fu_elantp_hid_device_setup(FuDevice *device, GError **error)
 	if (!fu_elantp_hid_device_ensure_iap_ctrl(self, error))
 		return FALSE;
 
+	if (fu_elantp_hid_device_read_hatpic_enable(self, error)) {
+		g_autoptr(FuElantpHidHapticDevice) cfg = fu_elantp_haptic_device_new(device);
+		fu_device_add_child(FU_DEVICE(device), FU_DEVICE(cfg));
+	}
```

Comment:
you need to deal with the GError, e.g.

	g_autoptr(GError) error_local = NULL;
	if (!fu_elantp_hid_device_read_hatpic_enable(self, &error_local)) {
		g_debug("no haptic device detected: %s", error_local->message);
	} else {
		g_autoptr(FuElantpHidHapticDevice) cfg = fu_elantp_haptic_device_new(device);
		fu_device_add_child(FU_DEVICE(device), FU_DEVICE(cfg));
	}